File: back-end/wlms/src/main/java/com/br/motorola/wlms/model/TaskSprint.java
package com.br.motorola.wlms.model;

import java.sql.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import lombok.*;

@Getter
@Setter
@Entity
public class TaskSprint {
  

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(nullable = false, updatable = false)
        private Long task_sprint_id; 

        @Column(nullable = false)
        private String task_sprint_name;

        @Column(nullable = false)
        private String task_sprint_description; 

        @Column(nullable = false)
        private Date task_sprint_start_date;

        @Column(nullable = false)
        private Date task_sprint_end_date;

        @Column(nullable = false)
        private String task_sprint_description_conflict;

        @Column(nullable = false)
        private double task_sprint_completion_level;

        @Column(nullable = false)
        private String task_sprint_responsible_ids;

        @Column(nullable = false)
        private String task_sprint_responsible_conflict_ids;

        @ManyToOne(optional = false)
        @JoinColumn(name = "task_sprint_sprint_fk")
        private Sprint sprint;
        
             
}

File: front-end/sgc/src/app/model/taskSprint.ts
import { Sprint } from "./sprint"

export interface TaskSprint {

  task_sprint_id: number;
  task_sprint_name: string;
  task_sprint_description: string;
  task_sprint_start: string;
  task_sprint_end: string;
  task_sprint_description_conflict: string;
  task_sprint_completion_level: number; // este era double no bd.
  task_sprint_responsible_ids: string; 
  task_sprint_responsible_conflict_ids: string; 
  sprint: Sprint; //fk

}
